codebuild/bin/s2n_dynamic_load_test.c

```diff
@@ -63,17 +63,22 @@ static void *s2n_load_dynamic_lib(void *ctx)
         fprintf(stderr, "Error calling s2n_init: '%s'\n", (*s2n_strerror_debug_dl)(s2n_errno, "EN"));
         exit(1);
     }
-    if ((*s2n_cleanup_dl)()) {
+    if ((*s2n_cleanup_final_dl)()) {
         int s2n_errno = (*s2n_errno_location_dl)();
-        fprintf(stderr, "Error calling s2n_cleanup: '%s'\n", (*s2n_strerror_debug_dl)(s2n_errno, "EN"));
+        fprintf(stderr, "Error calling s2n_cleanup_final: '%s'\n", (*s2n_strerror_debug_dl)(s2n_errno, "EN"));
         exit(1);
     }
 
+    /* TODO: https://github.com/aws/s2n-tls/issues/4827
+     * This is a bug. We can get this test to
+     * pass by commenting out dlclose, however this issue eventually
+     * needs to be fixed.
```

Comment:
Reading the issue, I'm not sure I understand how this is fixed by not calling `dlclose`? Is `dlclose` what invokes the process-exit destructor?

Reply:
Good callout! I forget! I clearly need a better comment :) Let me go remember what this was...